lib/elixir/src/elixir_parser.yrl

```diff
@@ -736,16 +739,29 @@ build_sigil({sigil, Location, Sigil, Parts, Modifiers, Terminator}) ->
   MetaWithTerminator = [{terminator, Terminator} | Meta],
   {list_to_atom("sigil_" ++ [Sigil]), MetaWithTerminator, [{'<<>>', Meta, string_parts(Parts)}, Modifiers]}.
 
-build_bin_string({bin_string, _Location, [H]} = Token) when is_binary(H) ->
-  handle_literal(H, Token);
-build_bin_string({bin_string, Location, Args}) ->
-  {'<<>>', meta_from_location(Location), string_parts(Args)}.
+build_bin_heredoc({bin_heredoc, Location, Args}) ->
+  build_bin_string({bin_string, Location, Args}, [{format, bin_heredoc}]).
 
-build_list_string({list_string, _Location, [H]} = Token) when is_binary(H) ->
-  handle_literal(elixir_utils:characters_to_list(H), Token);
-build_list_string({list_string, Location, Args}) ->
+build_list_heredoc({list_heredoc, Location, Args}) ->
+  build_list_string({list_string, Location, Args}, [{format, list_heredoc}]).
+
+build_bin_string(Token) ->
+  build_bin_string(Token, []).
+
+build_bin_string({bin_string, _Location, [H]} = Token, ExtraMeta) when is_binary(H) ->
+  handle_literal(H, Token, ExtraMeta);
+build_bin_string({bin_string, Location, Args}, ExtraMeta) ->
+  Meta = meta_from_location(Location) ++ ExtraMeta,
+  {'<<>>', Meta, string_parts(Args)}.
```

Comment:
Ok, so to clarify, it only makes sense to put in the front because almost always it is smaller than the `meta_from_location(Location)`. :)